app/lib/screens/authenticate/sign_in.dart

```diff
@@ -31,18 +32,14 @@ class SignInScreen extends StatelessWidget {
                   controller: passwordController,
                   decoration: InputDecoration(hintText: "Password"),
                 ),
-                // TODO: Add logic that will disable the widget when we're awaiting for response
-                model.viewState == ViewState.Busy
-                    ? CircularProgressIndicator()
-                    : RaisedButton(
-                        onPressed: () {
-                          model.signIn(
-                              email: emailController.text.trim(),
-                              password: passwordController.text.trim());
-                        },
-                        child: Text('Sign In'),
-                        color: Colors.green,
-                      ),
+                RaisedButton(
+                  onPressed: () async => LoadingOverlayScreen.of(context)
```

Comment:
Sign in behaviour has not changed, once the work is done the loading overlay screen is popped and the new screen navigated to without conflict.